consensus/src/main/java/org/ethereum/beacon/consensus/SpecHelpers.java

```diff
@@ -51,9 +55,24 @@
  */
 public class SpecHelpers {
   private final ChainSpec spec;
+  private final Hasher<Hash32> objectHasher;
 
+  /* Uses Hash32.ZERO as a stub for objectHasher */
   public SpecHelpers(ChainSpec spec) {
     this.spec = spec;
+    this.objectHasher = input -> Hash32.ZERO;
+  }
+
+  /** Builds objectHasher with {@link #hash(BytesValue)} as data hash function in objectHasher */
+  public SpecHelpers(
+      ChainSpec spec, Function<Function<BytesValue, Hash32>, Hasher<Hash32>> objectHasherBuilder) {
+    this.spec = spec;
+    this.objectHasher = objectHasherBuilder.apply(this::hash);
```

Comment:
Could we explicitly call `SSZHasher.createXXX(Function<BytesValue, Hash32> hashFunction)` instead of this? What could the other application for this abstraction be?

Reply:
how could we pass this::hash which is non static this way?

Reply:
It could be passed as an argument to a factory method, I think.

Reply:
Something like this:
```this.objectHasher = SSZHasher.createXXX(this::hash);```